test/fork/ConstantProductHelper.t..sol

```diff
@@ -0,0 +1,183 @@
+        assertGe(WETH.balanceOf(address(settlement)), wethBuffer);
+
+        // Verify that the tokens are the expected ones.
+        IERC20[] memory tokens = addressVecToIerc20Vec(helper.tokens(wethUsdcAmm));
+        uint256 usdcIndex = 0;
+        uint256 wethIndex = 1;
+        assertEq(tokens.length, 2);
+        assertEq(address(tokens[usdcIndex]), address(USDC));
+        assertEq(address(tokens[wethIndex]), address(WETH));
+
+        // Prepare the price vector used in the execution of the settlement in
+        // CoW Protocol. We skew the price by ~5% towards a cheaper WETH, so
+        // that the AMM wants to buy WETH.
+        uint256[] memory prices = new uint256[](2);
+        // Note: oracle price are expressed in the same format as prices in
+        // a call to `settle`, where the  price vector is expressed so that
+        // if the first token is DAI and the second WETH then a price of 3000
+        // DAI per WETH means a price vector of [1, 3000] (if the decimals are
+        // different, as in WETH/USDC, then the atom amount is what counts).
+        prices[usdcIndex] = WETH.balanceOf(wethUsdcAmm);
+        prices[wethIndex] = USDC.balanceOf(wethUsdcAmm) * 95 / 100;
+
+        // Prepare the vector storing the one trade to be used in the
+        // settlement.
+        GPv2Trade.Data[] memory trades = new GPv2Trade.Data[](1);
+
+        // The helper generates the AMM order
+        GPv2Order.Data memory ammOrder;
+        GPv2Interaction.Data[] memory preInteractions;
+        GPv2Interaction.Data[] memory postInteractions;
+        bytes memory sig;
+        (ammOrder, preInteractions, postInteractions, sig) = helper.order(wethUsdcAmm, prices);
+        // The signature is valid for the contract but not for the
+        // settlement contract. We need to prepend the verifying contract
+        // address.
+        // DISCUSS: should we change that?
+        sig = abi.encodePacked(wethUsdcAmm, sig);
+        trades[0] = orderToFullTrade(ammOrder, tokens, GPv2Signing.Scheme.Eip1271, sig);
+
+        // We expect a commit interaction in both pre and post interactions
+        assertEq(preInteractions.length, 1);
+        assertEq(postInteractions.length, 1);
+
+        // Because of how we changed the price, we expect to buy USDC
+        assertEq(address(ammOrder.sellToken), address(USDC));
+        assertEq(address(ammOrder.buyToken), address(WETH));
+        // Check that the amounts and price aren't unreasonable. We changed the
+        // price by about 5%, so the amounts aren't expected to change
+        // significantly more (say, they are between 2% and 3% of the original
+        // balance).
+        assertGt(ammOrder.sellAmount, ammUsdcInitialBalance * 2 / 100);
+        assertLt(ammOrder.sellAmount, ammUsdcInitialBalance * 3 / 100);
+        assertGt(ammOrder.buyAmount, ammWethInitialBalance * 2 / 100);
+        assertLt(ammOrder.buyAmount, ammWethInitialBalance * 3 / 100);
```

Comment:
uber nit: could use [`assertApproxEqRel`](https://book.getfoundry.sh/reference/forge-std/assertApproxEqRel)

Reply:
could this test do an exact expectation? 

i find 2 paths for this:
- used forked balances, explain them, explain the math and do an `assertEq` with a value extracted from the test itself (e.g. 3045.12321321)
- replace forked balances with mocked ones (e.g. pool initial balance is `30.000:10`), calculate the expected outcome, do an `assertApproxEqRel` with a very tight tolerance

Reply:
> uber nit: could use [assertApproxEqRel](https://book.getfoundry.sh/reference/forge-std/assertApproxEqRel)

https://github.com/cowprotocol/cow-amm/pull/92/commits/b8369657077b679f0fbe7990aba115296414853b :heavy_check_mark: 

> could this test do an exact expectation?

There has been quite a lot of internal discussion on this so the logic used to be somewhat open when the test was written. But I believe the changes resulting from #94 will make everything easier: it just means that the order should use the lowest possible buy amount for the AMM. Will follow up if I get a satisfactory confirmations (maybe @acanidio-econ can chime in?).

Reply:
Given a sell amount, the (limit) buy amount should be set to the smallest possible buy amount. Unfortunately, I am not familiar with the code much. If I understand correctly, a constant product pool should (up to rounding) return `buy_amount = ammWethInitialBalance * sell_amount / (ammUsdcInitialBalance - sell_amount)`.

For general pools, it should return whatever the pool would give as out amount to a user when the in amount from the user is fixed, in a setting with zero fees.

Reply:
I agree with @fedgiac and @fhenneke : the order should use the lowest possible buy amount for the AMM